src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1238,14 +1316,39 @@ private void doNetworkSignal(
         private void doEncodePublish(
             long traceId,
             long authorization,
-            String topicName,
+            int subscriptionId,
+            OctetsFW extension,
             OctetsFW payload)
         {
+            final MqttDataExFW dataEx = extension.get(mqttDataExRO::tryWrap);
+            final StringFW topic = dataEx.topic();
+            final String topicName = topic.asString();
+
+            if (subscriptionId > 0)
+            {
+                mqttPropertyRW.wrap(mqttPropertyBuffer, 0, mqttPropertyBuffer.capacity())
+                    .subscriptionId(v -> v.set(subscriptionId));
+            }
+            mqttPropertyRW.wrap(mqttPropertyBuffer, mqttPropertyRW.limit(), mqttPropertyBuffer.capacity())
+                .messageExpiryInterval(dataEx.expiryInterval())
+                .wrap(mqttPropertyBuffer, mqttPropertyRW.limit(), mqttPropertyBuffer.capacity())
+                .contentType(dataEx.contentType().asString())
+                .wrap(mqttPropertyBuffer, mqttPropertyRW.limit(), mqttPropertyBuffer.capacity())
+                .payloadFormatIndicator((byte) dataEx.format().get().ordinal())
+                .wrap(mqttPropertyBuffer, mqttPropertyRW.limit(), mqttPropertyBuffer.capacity())
+                .responseTopic(dataEx.responseTopic().asString())
+                .wrap(mqttPropertyBuffer, mqttPropertyRW.limit(), mqttPropertyBuffer.capacity())
+                .correlationData(a -> a.bytes(dataEx.correlationInfo().bytes()))
+                .build();
```

Comment:
This is a little difficult to read, though logically correct.

Please split each property into a separate wrap + set + build to improve readability.

```
MqttPropertyFW.Builder mqttProperty = mqttPropertyRW.wrap(mqttPropertyBuffer, 0, mqttPropertyBuffer.capacity());

if (subscriptionId > 0)
{
    mqttProperty.wrap(mqttPropertyBuffer, mqttProperty.limit(), mqttPropertyBuffer.capacity())
                .subscriptionId(v -> v.set(subscriptionId));
}
mqttProperty.wrap(mqttPropertyBuffer, mqttProperty.limit(), mqttPropertyBuffer.capacity())
            .messageExpiryInterval(dataEx.expiryInterval())
            .build()
mqttProperty.wrap(mqttPropertyBuffer, mqttProperty.limit(), mqttPropertyBuffer.capacity())
            .contentType(dataEx.contentType().asString())
            .build()
...
```
Also, check if it is necessary to call `asString()`, would be better to avoid `UTF-8` decoding and encoding if not needed.